src/rgw/rgw_lua_utils.h

```diff
@@ -201,7 +204,11 @@ int StringMapWriteableNewIndex(lua_State* L) {
 
   const char* index = luaL_checkstring(L, 2);
   const char* value = luaL_checkstring(L, 3);
-  map->insert_or_assign(index, value);
+  if (strlen(value) < MAX_LUA_VALUE_SIZE && map->size() < MAX_LUA_KEY_ENTRIES) {
```

Comment:
* please check seperatly, so you can give the correct error message
* please use `strnlen()` (`strlen()` is unsafe)
* please check that length of value + key don't exceed the max (otherwise users can "hack" the limit by putting the value in the key :-)